### ISSUE TITLE
Transaction level pooling

### ISSUE THREAD

**Issue by knizhnik** [agroal/pgagroal#39]
As far as I understand pgagroal implements session level pooling, i.e. number of sessions is equal to number of Postgres backends.
So  pgagroal can not be used to restrict number of Postgres backends (as pgbouncer/Odyssey can do). It can only help when clients do not establish permanent connections with database but perform connect/disconnect on each operation. But it seems to be quite exotic use case now, doesn't it?

Are there any plans to support transaction level pooling?

**Comment by jesperpedersen** [comment]
Yes, currently `pgagroal` is session based.

I don't see any reason why there shouldn't be a transaction based model though, so I have created [#40] for this. I have left it a community issue for now, as I got other stuff I need to look at first.

See the pool-less mode of https://agroal.github.io/ to see an example of where a connection is released on a transaction boundary.

If you are happy with this answer feel free close this issue.

Thanks for your feedback Konstantin !
